builtin/providers/aws/resource_aws_s3_bucket_object.go

```diff
@@ -218,7 +229,11 @@ func resourceAwsS3BucketObjectRead(d *schema.ResourceData, meta interface{}) err
 	d.Set("content_language", resp.ContentLanguage)
 	d.Set("content_type", resp.ContentType)
 	d.Set("version_id", resp.VersionId)
-	d.Set("kms_key_id", resp.SSEKMSKeyId)
+	if _, ok := d.GetOk("server_side_encryption"); ok {
+		d.Set("server_side_encryption", resp.ServerSideEncryption)
+	} else if _, ok := d.GetOk("kms_key_id"); ok {
+		d.Set("kms_key_id", resp.SSEKMSKeyId)
```

Comment:
@radeksimko thanks for the feedback. That makes sense. I figured there was a better way of handling this. My grasp on the colloquial TF knowledge was too thin. I'll give those changes a go.